lib/srv/db/common/errors.go

```diff
@@ -102,3 +104,57 @@ type causer interface {
 type pgError interface {
 	Unwrap() error
 }
+
+// ConvertConnectError converts common connection errors to trace errors with
+// extra information/recommendations if necessary.
+func ConvertConnectError(err error, sessionCtx *Session) error {
+	errString := err.Error()
+	switch {
+	case strings.Contains(errString, "x509: certificate signed by unknown authority"):
```

Comment:
Can we leverage the `x509.UnknownAuthorityError` error type here instead of string base comparison ?

Reply:
database libs/drivers may have their extra message on top of the original x509 error, like `failed to write startup message (x509: certificate signed by unknown authority)`